core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -370,6 +370,14 @@ object SparkHadoopUtil {
 
   val SPARK_YARN_CREDS_COUNTER_DELIM = "-"
 
+  /**
+   * Number of records to update input metrics when reading from HadoopRDDs.
+   *
+   * Each update is potentially expensive because we need to use reflection to access the
+   * Hadoop FileSystem API of interest (only available in 2.5), so we should do this sparingly.
+   */
+  private[spark] val UPDATE_INPUT_METRICS_INTERVAL_RECORDS = 1000
```

Comment:
we used to do this as part of the confusing bytes callback in `InputMetrics`. The purpose of that was to keep the UI updated as we read input data. This is just a more straightforward way of doing it so we can keep `InputMetrics` simple.